src/edu/psu/sweng500/team8/play/ActionManager.java

```diff
@@ -11,20 +11,20 @@
 	 * so they can be adjusted on the fly. 
 	 * Or we can still include it in the JAR but as a separate file
 	 * so we can change it through the development process.  */
-	private static final int QUEUE_SIZE_LIMIT = 10;
+	private static final int STACK_SIZE_LIMIT = 10;
 	
-	private Stack<SudokuAction> sudokuActionQueue;
-	private Stack<SudokuAction> sudokuActionQueueForUndo;
+	private Stack<SudokuAction> sudokuActionStack;
+	private Stack<SudokuAction> sudokuActionStackForUndo;
 
 	public ActionManager() {
-		this.sudokuActionQueue = new Stack<SudokuAction>();
-		this.sudokuActionQueue.setSize(QUEUE_SIZE_LIMIT);		
-		this.sudokuActionQueueForUndo = new Stack<SudokuAction>();
-		this.sudokuActionQueueForUndo.setSize(QUEUE_SIZE_LIMIT);
+		this.sudokuActionStack = new Stack<SudokuAction>();
+		this.sudokuActionStack.setSize(STACK_SIZE_LIMIT);		

```

Comment:
I don't think we need to restrict the size. If we do, your current implementation is not restricting it anyway. Stack will automatically resize if you go add more than the current size. The problem is that it is slow because it has to copy elements to the new vector. If you want to set the initial size to minimize resizing, use the ensureCapacity function.